dartagnan/src/main/java/com/dat3m/dartagnan/wmm/RelationNameRepository.java

```diff
@@ -13,6 +13,7 @@ public class RelationNameRepository {
     public static final String EXT = "ext";
     public static final String CO = "co";
     public static final String RF = "rf";
+    public static final String UR = "UR";
```

Comment:
isn't `UR` a set and `[UR]` the relation?

Reply:
We cannot define dynamic sets (see #371 ) right now, so I had to define `UR` to be the set identity (binary relation) on uninitialized reads. That's why the derived relations use `UR` rather than `[UR]`.

Reply:
I know we cannot yet handle unary predicates. However, isn't the situation and the problem exactly the same that we have with e.g., `[range(r)]`? If so, we should use `[UR]` even if we are "hardcoding the relation" rather than creating it from a filter.

Reply:
No, it's not the same situation. In the case of `[range(r)]` the trick was to define the combined operator `[range( . )]` directly which also means that the parser has a special rule for that.
To do the same with `[UR]` you would need to change the grammar (or at least the parser) to have a special case just for that one relation. I think that is not a good idea because it is super ambiguous.

Btw. I could also avoid a new base relation and instead define `UR = [R] \ [range(rf)]`. I guess there is not much difference in encoding and verification performance.

Reply:
I replaced `UR` by `[R] \ [range(rf)]` to simplify the code even more.